plugins/inputs/p4runtime/p4runtime_test.go

```diff
@@ -0,0 +1,655 @@
+package p4runtime
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	p4ConfigV1 "github.com/p4lang/p4runtime/go/p4/config/v1"
+	p4v1 "github.com/p4lang/p4runtime/go/p4/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+// CounterSpec available here https://github.com/p4lang/p4runtime/blob/main/proto/p4/config/v1/p4info.proto#L289
+func createCounter(
+	name string,
+	id uint32,
+	unit p4ConfigV1.CounterSpec_Unit,
+) *p4ConfigV1.Counter {
+	return &p4ConfigV1.Counter{
+		Preamble: &p4ConfigV1.Preamble{Name: name, Id: id},
+		Spec:     &p4ConfigV1.CounterSpec{Unit: unit},
+	}
+}
+
+func createTelegrafMetric(
+	tags map[string]string,
+	fields map[string]interface{},
+) telegraf.Metric {
+	return testutil.MustMetric("p4_runtime", tags, fields, time.Unix(0, 0))
+}
```

Comment:
Does that really save lines? How about unroll this indirection and use `testutil.MustMetric` or even `metric.New` at the corresponding places?

Reply:
Only a few, so I unrolled this method everywhere